python/ray/tests/test_symbol_collisions.py

```diff
@@ -0,0 +1,29 @@
+"""
+This script ensures that various libraries do not conflict with ray by
+trying to import both libraries in both orders.
+A specific example is that importing ray after pyarrow causes a Segfault.
+"""
+import subprocess
+
+TESTED_LIBRARIES = ["pyarrow"]
```

Comment:
I'm mildly surprised that this test passes in our CI. Is pyarrow installed in our CI?

Reply:
This doesn't fail if `pyarrow` is installed, only if there is some strange symbol collision. There was initially a fix to limit the scope of exported symbols, but that was removed after it became un-reproducable.

Reply:
Right, I agree it doesn't fail if `pyarrow` is installed. I just thought that `pyarrow` isn't installed in our Travis tests.

Reply:
Ohh okay! Let me check that.

Reply:
I added Pyarrow to the dependencies. It looks like the error is a segfault again :/ 
https://travis-ci.com/github/ray-project/ray/jobs/308220125

Reply:
Is this pyarrow version  0.14?

Reply:
No, it is 0.16. I added the version to the debug info, so it shows up in this [test](https://travis-ci.com/github/ray-project/ray/jobs/308290546).